Dell/recovery_backend.py

```diff
@@ -1013,6 +1014,18 @@ def create_ubuntu(self, recovery, revision, iso, platform, no_update, sender=Non
             xorrisoargs.append('boot/efi.img')
             xorrisoargs.append('-no-emul-boot')
             xorrisoargs.append('-isohybrid-gpt-basdat')
+        elif os.path.exists(os.path.join(mntdir, 'boot', '_efi.img')):
+            xorrisoargs.append('-partition_offset')
+            xorrisoargs.append('16')
+            xorrisoargs.append('-append_partition')
+            xorrisoargs.append('2')
+            xorrisoargs.append('0xEF')
+            xorrisoargs.append(os.path.join(mntdir, 'boot', '_efi.img'))
+            xorrisoargs.append('-appended_part_as_gpt')
+            xorrisoargs.append('-eltorito-alt-boot')
+            xorrisoargs.append('-e')
+            xorrisoargs.append('--interval:appended_partition_2:all::')
+            xorrisoargs.append('-no-emul-boot')
```

Comment:
I think it would be more scalable if we could build it as the way that the `efi.img` used in 1011-1016 worked.  Is there a reason it can't be?

Reply:
The partition table is missing from within the generated iso file, I think `-isohybrid-gpt-basdat` works only if `-isohybrid-mbr` is too used as xorriso's [manpage](https://www.gnu.org/software/xorriso/man_1_xorrisofs.html) described below:

> **-isohybrid-gpt-basdat**
> Mark the current El Torito boot image (see options -b and -e) in an actually invalid GPT as partition of type Basic Data. This works only with -isohybrid-mbr and has the same impact on the system area as -efi-boot-part. 
> 
> 
With Ubuntu Desktop 20.10 onward the isolinux is excluded from the generic iso file therefore the generated iso doesn't carry MBR.

Reply:
Ah I see... OK.